pkg/controller/clustersync/clustersync_controller_test.go

```diff
@@ -1817,6 +1817,49 @@ func TestReconcileClusterSync_NoFirstSuccessTimeSet(t *testing.T) {
 	assert.Nil(t, updatedClusterSync.Status.FirstSuccessTime)
 }
 
+func TestReconcileClusterSync_UpsertResourcesToDelete(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	scheme := newScheme()
+	resourcesToApply := testConfigMap("dest-namespace", "dest-name")
+	syncSet := testsyncset.FullBuilder(testNamespace, "test-syncset", scheme).Build(
+		testsyncset.ForClusterDeployments(testCDName),
+		// ResourceApplyMode is Upsert
+		testsyncset.WithApplyMode(hivev1.UpsertResourceApplyMode),
+		testsyncset.WithGeneration(2),
+		testsyncset.WithResources(resourcesToApply),
+	)
+	existingSyncStatus := buildSyncStatus("test-syncset",
+		withTransitionInThePast(),
+		withFirstSuccessTimeInThePast(),
+		// Existing SyncStatus has ResourcesToDelete
+		withResourcesToDelete(testConfigMapRef("dest-namespace", "dest-resource")),
```

Comment:
If this is changed to the correct name of `dest-name` instead of `dest-resource`, this test fails.